homeassistant/components/radarr/coordinator.py

```diff
@@ -166,39 +167,43 @@ async def async_get_events(
     ) -> list[RadarrEvent]:
         """Get cached events and request missing dates."""
         # remove older events to prevent memory leak
+        start = start_date.date()
         self._events = [
-            e
-            for e in self._events
-            if e.start >= datetime.now().date() - timedelta(days=30)
+            e for e in self._events if e.start >= dt_util.now() - timedelta(days=30)
         ]
-        _days = (end_date - start_date).days
+        _days = (end_date.date() - start).days
         await asyncio.gather(
             *(
                 self._async_get_events(d)
-                for d in ((start_date + timedelta(days=x)).date() for x in range(_days))
-                if d not in (event.start for event in self._events)
+                for d in (start + timedelta(days=x) for x in range(_days))
+                if d
+                not in (cast(datetime, event.start).date() for event in self._events)
             )
         )
         return self._events
 
     async def _async_get_events(self, _date: date) -> None:
         """Return events from specified date."""
+        info = dt_util.get_default_time_zone()
+        offset = info.utcoffset(dt_util.now())
```

Comment:
This should no longer be needed now? This is the whole thing i want to get rid of.

If you make sure it is the right timezone, start of local day should get this right directly.